mavros/src/plugins/setpoint_attitude.cpp

```diff
@@ -215,6 +222,53 @@ class SetpointAttitudePlugin : public plugin::PluginBase,
 			send_attitude_ang_velocity(req->header.stamp, ang_vel, thrust_msg->thrust);
 	}
 
+	/**
+	 * @brief Setpoint for Roll, Pitch, Yawrate and Thrust (RPYT).
+	 * Message specification: http://docs.ros.org/indigo/api/mav_msgs/html/msg/RollPitchYawrateThrust.html
+	 * Thrust is set in Newtons and scaled by 1/MAX_THRUST_N (MAX_THRUST_N is UAV dependent).
+	 * @param msg		Received RPYT setpoint
+	 */
+	void rpyt_cb(const mav_msgs::RollPitchYawrateThrust::ConstPtr msg)
+	{
+
+		// Set Thrust scaling in px4_config.yaml, setpoint_attitude block.
+		if(!sp_nh.getParam("thrust_scaling", thrust_scaling)){
+			ignore_rpyt_messages_ = true;
+		}
+		
+		if(ignore_rpyt_messages_){
+			ROS_FATAL("Recieved roll_pitch_yaw_thrust_rate message, but ignore_rpyt_messages_ is true: "
+	 	        "the most likely cause of this is a failure to specify the thrust_scaling parameters "
+	 	        "on px4/apm_config.yaml .");
+		     	return;
+		}
+
+		if(thrust_scaling == 0.0){
+			ROS_WARN("thrust_scaling parameter is set to zero.");
+		}
+
+		// Set mask to ignore everything but thrust and attitude setpoint
+		uint8_t ignore_all_except_q_and_thrust = 0; //(7 << 0);
+
+		// Transforms from thrust acceleration to a thrust force (scaling adjusts to the UAV
```

Comment:
Comment is incorrect. It is already a force.

Reply:
"Normalizes thrust force (in Newtons) to a range of 0-1" is more appropriate.

Reply:
True, I changed to zero just for trying, forget to put it back. Will do soon :)